paymentsheet/src/test/java/com/stripe/android/paymentsheet/state/CustomerStateTest.kt

```diff
@@ -112,6 +114,45 @@ class CustomerStateTest {
                     // Always true for `customer_session`
                     canRemoveDuplicates = true,
                 ),
+                defaultPaymentMethodId = null
+            )
+        )
+    }
+
+    @Test
+    fun `Should create 'CustomerState' for customer session properly with nonnull defaultPaymentMethodId`() {
+        val paymentMethods = PaymentMethodFactory.cards(3)
+        val customer = createElementsSessionCustomer(
+            customerId = "cus_3",
+            ephemeralKeySecret = "ek_3",
+            paymentMethods = paymentMethods,
+            mobilePaymentElementComponent = ElementsSession.Customer.Components.MobilePaymentElement.Enabled(
+                isPaymentMethodSaveEnabled = false,
+                isPaymentMethodRemoveEnabled = false,
+                canRemoveLastPaymentMethod = false,
+                allowRedisplayOverride = null,
+            ),
+            defaultPaymentMethodId = "aaa111"
+        )
+
+        val customerState = CustomerState.createForCustomerSession(
+            customer = customer,
+            configuration = createConfiguration(),
+            supportedSavedPaymentMethodTypes = listOf(PaymentMethod.Type.Card)
+        )
+
+        assertThat(customerState).isEqualTo(
```

Comment:
Tip for the future (doesn't need to be changed in this PR): I generally prefer asserting equality of individual fields in these tests instead of asserting equality of the whole object. A few reasons:
- it makes it more obvious what specifically is being tested, e.g. in this test we are testing the defaultPaymentMethodId behavior, so we can just test that that field is correct
- it makes it easier to read test failures in the future. If this test fails as-is, the assertion error message will be quite long and show two different objects. If you assert on individual fields, you will know immediately which field is different and how it is different.
- it requires us to update fewer tests in the future if we add more fields. E.g. in this PR, you needed to add `defaultPaymentMethodId = null` to multiple existing tests. If we only asserted on individual fields, our existing tests would compile + pass without any action on your part